pallets/liquid-staking-two/src/lib.rs

```diff
@@ -0,0 +1,820 @@
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn deposit_event_to_relaychain(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+            //todo ensure that this method is called after update era index.
+            let who = ensure_signed(origin)?;
+            // match stake and unstake,
+            let previous_era = Self::previous_era();
+            let pool_buffer = MatchingPool::<T>::get(&previous_era);
+            let pool_stake_amount = pool_buffer.total_stake_amount;
+            let pool_unstake_amount = pool_buffer.total_unstake_amount;
+            let (operation_type, amount) = if pool_stake_amount > pool_unstake_amount {
+                (
+                    StakingOperationType::Bond,
+                    pool_stake_amount - pool_unstake_amount,
+                )
+            } else if pool_stake_amount < pool_unstake_amount {
+                (
+                    StakingOperationType::Unbond,
+                    pool_unstake_amount - pool_stake_amount,
+                )
+            } else {
+                (StakingOperationType::Matching, 0)
+            };
+            StakingOperationHistory::<T>::try_mutate(
+                &previous_era,
+                &operation_type,
+                |o| -> DispatchResult {
+                    ensure!(*o == None, "error");
+                    o.as_mut().and_then(|operation| {
+                        operation.status = ResponseStatus::Ready;
+                        operation.amount = amount;
+                        operation.block_number = frame_system::Pallet::<T>::block_number();
+                        Some(())
+                    });
+
+                    Ok(())
+                },
+            )?;
+            MatchingPool::<T>::try_mutate(&previous_era, |pool_buffer| -> DispatchResult {
+                pool_buffer.operation_type = Some(operation_type);
+                Ok(())
+            })?;
+            // Deposit event, Offchain stake-client need to listen this
+            Self::deposit_event(Event::DepositEventToRelaychain(
+                who,
+                previous_era,
+                operation_type,
+                amount,
+            ));
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn stake(
+            origin: OriginFor<T>,
+            #[pallet::compact] amount: Balance,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+            <Self as LiquidStakingProtocol<T::AccountId>>::stake(&who, amount)?;
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn unstake(
+            origin: OriginFor<T>,
+            #[pallet::compact] amount: Balance,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+            <Self as LiquidStakingProtocol<T::AccountId>>::unstake(&who, amount)?;
```

Comment:
DispatchResult should be preferred if we don't return anything

Reply:
Same upon.